tinygrad/graph.py

```diff
@@ -86,16 +86,27 @@ def log_lazybuffer(lb, scheduled=False):
       if nm(lb) not in G.nodes:
         # realized but unseen?
         G.add_node(nm(lb), label=f'"{str(lb.base.realized)[5:-1].replace(" ", chr(10))}\nb:{bm(lb.realized)}"', style='filled', fillcolor="#f0c08080")
+
+cycle_tracker: Dict[int, Tuple[int, int]] = {}
```

Comment:
Why not pass these in to the _tree function? I did the other caches without globals